### ISSUE TITLE
General solution to specific webpack configs in CT

### ISSUE THREAD

**Issue by lmiller1990** [cypress-io/cypress#16509]
### Current behavior

Need some way to support webpack configurations designs for specific environments.

-  [CRA module mapper problems](https://github.com/cypress-io/cypress/discussions/15895). CRA does not let you transpile TS outside of `src`.
- [Next.js SCSS problem](https://github.com/cypress-io/cypress/discussions/16294#discussioncomment-726530)
-  [CRA Fork Plugin problems](https://github.com/cypress-io/cypress/issues/16169](https://github.com/cypress-io/cypress/issues/16169)
- Likely Angular will have similar problems

The common issue is those webpack configs are expected to be used in some specific scenario, and Cypress does not match those conditions. We need some kind of general solution for this.

### Desired behavior

It "just works". Under the hood can be ugly, as long as the user experience is smooth.

### Test code to reproduce

See above issues.

**Comment by lmiller1990** [comment]
Related: configuing global styles, see comment: https://github.com/cypress-io/cypress/discussions/16294#discussioncomment-813497

TL;DR configure global styles like this:

```js
// cypress/support/index.js

import { mount, configure } from '@cypres/react'

configure({
  globalStylesheets: [/* ... */]
})
```

**Comment by rockindahizzy** [comment]
We've refactored CT and overhauled how webpack is handled.  Going to close this one, we can reopen if its still an issue.
